src/controllers/mentionEachUser.ts

```diff
@@ -23,34 +23,36 @@ export async function mentionEachUser(
 ) {
   const getMembersInServerResponse = await getMembersInServer(env);
   const roleId = transformedArgument.roleToBeTaggedObj.value;
-  const msgToBeSent = transformedArgument?.displayMessageObj?.value;
+  const msgToBeSent = transformedArgument?.displayMessageObj?.value; // Get custom message
   const dev = transformedArgument?.dev?.value || false;
-  // optional chaining here only because display message obj is optional argument
+
   const usersWithMatchingRole = filterUserByRoles(
     getMembersInServerResponse as UserArray[],
     roleId
   );
-  const payload = {
-    channelId: transformedArgument.channelId,
-    roleId: roleId,
-    message: msgToBeSent,
-    usersWithMatchingRole,
-  };
+
+  // Use the custom message if provided, otherwise construct the default message
+  let responseMessage;
+  if (usersWithMatchingRole.length === 0) {
+    responseMessage = msgToBeSent || "Sorry no user found under this role."; // Use custom message if available
+  } else {
+    const roleTag = `<@&${roleId}>`;
+    const userList = usersWithMatchingRole.join(", ");
+    responseMessage = `The users with ${roleTag} roles are: ${userList}`;
+  }
+
```

Comment:
Why we doing this as you need to show only message of mention each for which role is selected

Reply:
I made changes to  ensure that the response message is returned  based on the presence or absence of users with the specified role.